packages/sign-client/src/client.ts

```diff
@@ -200,6 +230,42 @@ export class SignClient extends ISignClient {
     }
   };
 
+  public authenticate: ISignClient["authenticate"] = async (params) => {
+    try {
+      return await this.engine.authenticate(params);
+    } catch (error: any) {
+      this.logger.error(error.message);
+      throw error;
+    }
+  };
+
+  public formatAuthMessage: ISignClient["formatAuthMessage"] = (params) => {
+    try {
+      return this.engine.formatAuthMessage(params);
+    } catch (error: any) {
+      this.logger.error(error.message);
+      throw error;
+    }
+  };
+
+  public approveSessionAuthenticate: ISignClient["approveSessionAuthenticate"] = (params) => {
+    try {
+      return this.engine.approveSessionAuthenticate(params);
+    } catch (error: any) {
+      this.logger.error(error.message);
+      throw error;
+    }
+  };
+
+  public rejectSessionAuthenticate: ISignClient["rejectSessionAuthenticate"] = (params) => {
+    try {
+      return this.engine.rejectSessionAuthenticate(params);
+    } catch (error: any) {
+      this.logger.error(error.message);
+      throw error;
+    }
+  };
```

Comment:
It's missing `await` to catch the error.
```suggestion
  public approveSessionAuthenticate: ISignClient["approveSessionAuthenticate"] = (params) => {
    try {
      return await this.engine.approveSessionAuthenticate(params);
    } catch (error: any) {
      this.logger.error(error.message);
      throw error;
    }
  };

  public rejectSessionAuthenticate: ISignClient["rejectSessionAuthenticate"] = (params) => {
    try {
      return await this.engine.rejectSessionAuthenticate(params);
    } catch (error: any) {
      this.logger.error(error.message);
      throw error;
    }
  };
```